File: backend/Frametric.Application/DTOs/TmdbCollectionResultDto.cs
namespace Frametric.Application.DTOs;

public record TmdbCollectionResultDto(
    int Id,
    string Name,
    string? Overview,
    string? PosterUrl,
    string? BackdropUrl,
    List<TmdbCollectionPartDto> Parts
);

public record TmdbCollectionPartDto(
    int Id,
    string Title,
    string? ReleaseDate,
    string? PosterUrl,
    bool IsInDatabase
);

File: frontend/src/app/core/api/model/tmdb-collection-result-dto.ts
export interface TmdbCollectionResultDto {
    id: number;
    name: string;
    overview: string | null;
    posterUrl: string | null;
    backdropUrl: string | null;
    parts: Array<TmdbCollectionPartDto>;
}

export interface TmdbCollectionPartDto {
    id: number;
    title: string;
    releaseDate: string | null;
    posterUrl: string | null;
    isInDatabase: boolean;
}
